### PR TITLE
Backport of #9270: add Pyodide support and CI jobs for v0.14.x

### PR DESCRIPTION
**Pull request by agriyakhetarpal**
- [ ] closes #xxxx
- [x] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

This PR has been stemmed off from https://github.com/statsmodels/statsmodels/issues/9166#issuecomment-2352811196. It is a backport of #9270 to the [0.14.x branch](https://github.com/statsmodels/statsmodels/tree/maintenance/0.14.x) – the PR originally added Pyodide support and testing improvements through a CI job.

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by agriyakhetarpal** [comment]
[All tests are passing](https://github.com/statsmodels/statsmodels/actions/runs/10886832113/job/30207759234?pr=9365), I'll remove the workflow trigger.

File: .github/workflows/emscripten.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# Attributed to NumPy via https://github.com/numpy/numpy/pull/25894
+# https://github.com/numpy/numpy/blob/d2d2c25fa81b47810f5cbd85ea6485eb3a3ffec3/.github/workflows/emscripten.yml
+name: Test Emscripten/Pyodide build
+
+on:
+  push:
+    branches: [ 'main']
+  pull_request:
+    # TODO: remove when all tests pass and ready to merge
+    # branches: [ 'main' ]
```
Comment (agriyakhetarpal) [inline .github/workflows/emscripten.yml:10]:
```suggestion
  pull_request:
    branches: [ 'main' ]
```

**Comment by bashtage** [comment]
Thanks.

**Comment by agriyakhetarpal** [comment]
@bashtage, you mentioned a while earlier that a 0.14.4 release can be made with this change. Please let me know if you still have it in the ambit; I'm happy to help—where I can—with any other issues or PRs that might be needed for getting into a release.

**Comment by bashtage** [comment]
@agriyakhetarpal 0.14.4 has been released. The Pyodide wheel is here: https://anaconda.org/multibuild-wheels-staging/statsmodels/files?version=0.14.4

Thanks for all the work.

**Comment by agriyakhetarpal** [comment]
Superb, thank you for the review!

**Comment by agriyakhetarpal** [comment]
Superb, thank you again for all the reviews!

**Pull request merged**